pkg/driver/aws-ebs/aws_ebs_tags_retry_worker.go

```diff
@@ -0,0 +1,101 @@
+package aws_ebs
+
+import (
+	"context"
+	"errors"
+
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/klog/v2"
+
+	"github.com/openshift/library-go/pkg/controller/factory"
+)
+
+// startFailedQueueWorker runs a worker that processes failed volumes independently
+func (c *EBSVolumeTagsController) startFailedQueueWorker(ctx context.Context, syncContext factory.SyncContext) error {
+	for {
+		select {
+		case <-ctx.Done():
+			klog.Infof("Context canceled, stopping failed queue worker for EBS Volume Tags")
+			return errors.New("context canceled, stopping failed queue worker for EBS Volume Tags") // Stop the goroutine when the context is canceled
+		default:
+			// Get the next failed volume from the queue
+			item, quit := c.failedQueue.Get()
```

Comment:
This code is very hard to read, can you split this into multiple smaller functions please.